modules/programs/atuin.nix

```diff
@@ -94,49 +96,132 @@ in {
-      ${lib.getExe cfg.package} init fish ${flagsStr} | source
-    '';
-
-    programs.nushell = mkIf cfg.enableNushellIntegration {
-      extraEnv = ''
-        let atuin_cache = "${config.xdg.cacheHome}/atuin"
-        if not ($atuin_cache | path exists) {
-          mkdir $atuin_cache
-        }
-        ${
-          lib.getExe cfg.package
-        } init nu ${flagsStr} | save --force ${config.xdg.cacheHome}/atuin/init.nu
+      programs.zsh.initExtra = mkIf cfg.enableZshIntegration ''
+        if [[ $options[zle] = on ]]; then
+          eval "$(${lib.getExe cfg.package} init zsh ${flagsStr})"
+        fi
       '';
-      extraConfig = ''
-        source ${config.xdg.cacheHome}/atuin/init.nu
+
+      programs.fish.interactiveShellInit = mkIf cfg.enableFishIntegration ''
+        ${lib.getExe cfg.package} init fish ${flagsStr} | source
       '';
-    };
-  };
+
+      programs.nushell = mkIf cfg.enableNushellIntegration {
+        extraEnv = ''
+          let atuin_cache = "${config.xdg.cacheHome}/atuin"
+          if not ($atuin_cache | path exists) {
+            mkdir $atuin_cache
+          }
+          ${
+            lib.getExe cfg.package
+          } init nu ${flagsStr} | save --force ${config.xdg.cacheHome}/atuin/init.nu
+        '';
+        extraConfig = ''
+          source ${config.xdg.cacheHome}/atuin/init.nu
+        '';
+      };
+    }
+
+    (mkIf daemonCfg.enable (mkMerge [
+      {
+        assertions = [
+          {
+            assertion = versionAtLeast cfg.package.version "18.2.0";
+            message = ''
+              The Atuin daemon requires at least version 18.2.0 or later.
+            '';
+          }
+          {
+            assertion = isLinux || isDarwin;
+            message =
+              "The Atuin daemon can only be configured on either Linux or macOS.";
+          }
+        ];
+
+        programs.atuin.settings = { daemon = { enabled = true; }; };
+      }
+      (mkIf isLinux {
+        programs.atuin.settings = { daemon = { systemd_socket = true; }; };
+
+        systemd.user.services.atuin-daemon = {
+          Unit = {
+            Description = "Atuin daemon";
+            Requires = [ "atuin-daemon.socket" ];
+          };
+          Install = {
+            Also = [ "atuin-daemon.socket" ];
+            WantedBy = [ "default.target" ];
+          };
+          Service = {
+            ExecStart = "${lib.getExe cfg.package} daemon";
+            Environment = lib.optionals (daemonCfg.logLevel != null)
+              [ "ATUIN_LOG=${daemonCfg.logLevel}" ];
+            Restart = "on-failure";
+            RestartSteps = 3;
+            RestartMaxDelaySec = 6;
+          };
+        };
+
+        systemd.user.sockets.atuin-daemon = {
+          Unit = { Description = "Atuin daemon socket"; };
+          Install = { WantedBy = [ "sockets.target" ]; };
+          Socket = {
+            ListenStream = "%h/.local/share/atuin/atuin.sock";
```

Comment:
Shouldn't this use `%D` to refer to `$XDG_DATA_HOME`?

Reply:
We could well change this in the future. Perhaps you could make a PR to fix this, and link it to this PR?

Reply:
https://github.com/nix-community/home-manager/pull/6187

Reply:
Thanks 👍🏻